tensorboard/uploader/server_info.py

```diff
@@ -30,19 +31,31 @@
 _REQUEST_TIMEOUT_SECONDS = 10
 
 
-def _server_info_request():
+def _server_info_request(upload_plugins):
+    """Generates a ServerInfoRequest
+
+    Args:
+      upload_plugins: List of plugin names requested by the user and to be
+        verified by the server.
+
+    Returns:
+      A `server_info_pb2.ServerInfoRequest` message.
+    """
     request = server_info_pb2.ServerInfoRequest()
     request.version = version.VERSION
+    request.plugin_specification.upload_plugins[:] = upload_plugins
     return request
 
 
-def fetch_server_info(origin):
+def fetch_server_info(origin, upload_plugins):
     """Fetches server info from a remote server.
 
     Args:
       origin: The server with which to communicate. Should be a string
         like "https://tensorboard.dev", including protocol, host, and (if
         needed) port.
+      upload_plugins: List of plugins names requested by the user and to be
```

Comment:
Have you considered making `upload_plugins` a kwarg that defaults to `None`? That would eliminate many lines of change in the test file below.

Reply:
I think that this is fine as is, since we don’t want real callers to
accidentally forget to specify the list of plugins and `None` is not
currently a valid value here, anyway. But if @bmd3k wants to broaden
this, that’s fine with me, too.

Reply:
Right, callers in production are expected to set this value (for now).